### PR TITLE
feat(terraform): Refactor Module mapping objects

### PR DESCRIPTION
**Pull request by ChanochShayner**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description
Refactor - 
 - `module_dependency_map` - used in `get_module_vertices_mapping`, added support to this func without this object.
 - `_set_variables_values_from_modules` - Not used anywhere.
 - `dep_index_mapping` - No need.
 - `block.module_dependency` - add `source_module_object` instead.
 - `block.module_dependency_num` - add `source_module_object` instead.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/graph_builder/local_graph.py [hunk 1]
```diff
@@ -176,6 +139,21 @@ def get_module_vertices_mapping(self) -> None:
         For each vertex, if it's originated in a module import, add to the vertex the index of the
         matching module vertex as 'source_module'
         """
+        if strtobool(os.getenv('CHECKOV_NEW_TF_PARSER', 'False')):
+            for vertex in self.vertices:
+                if not vertex.source_module_object:
+                    continue
+                for i, source_v in enumerate(self.vertices):
```
Comment (achiar99) [inline checkov/terraform/graph_builder/local_graph.py:146]: what about the foreach index? @ChanochShayner

Comment (ChanochShayner) [reply checkov/terraform/graph_builder/local_graph.py:146]: We are not supporting the foreach index in modules for now.
When we will support Ill add it

**Review by achiar99 — APPROVED** [review]

File: checkov/terraform/graph_builder/local_graph.py [hunk 2]
```diff
@@ -176,6 +139,21 @@ def get_module_vertices_mapping(self) -> None:
         For each vertex, if it's originated in a module import, add to the vertex the index of the
         matching module vertex as 'source_module'
         """
+        if strtobool(os.getenv('CHECKOV_NEW_TF_PARSER', 'False')):
```
Comment (YaaraVerner) [inline checkov/terraform/graph_builder/local_graph.py:142]: can you save the value in the init function? like you did with `self.enable_foreach_handling`

File: checkov/terraform/graph_builder/local_graph.py (hunk 1 above)
Comment (YaaraVerner) [inline checkov/terraform/graph_builder/local_graph.py:146]:
```suggestion
                for i, source_v in enumerate(self.vertices_by_block_type[BlockType.MODULE]):
```
don't you need to go over only the modules blocks?

**Review by YaaraVerner — APPROVED** [review]
nice work!

**Pull request merged**